File: oops/solid/ts/S.ts

// Single Responsibility Principle

// following class violates the Single Responsibility Principle
class Student {
    public createStudentAccount() {
        // some logic
    }

    public calculateStudentGrade() {
        // some logic
    }

    public generateStudentData() {
        // some logic
    }
}

// Refactored code
class StudentAccount {
    public createStudentAccount() {
        // some logic
    }

    // other methods related to student account
}

class StudentGrade {
    public calculateStudentGrade() {
        // some logic
    }

    // other methods related to student grade
}

class StudentData {
    public generateStudentData() {
        // some logic
    }

    // other methods related to student data
}

File: oops/solid/py/S.py
"""
Single Responsibility Principle
"""

# following class violates the Single Responsibility Principle


class Student:

    def createStudentAccount():
        pass

    def calculateStudentGrade():
        pass

    def generateStudentData():
        pass


# correct way


class StudentAccount:

    def createStudentAccount():
        pass


class StudentGrade:

    def calculateStudentGrade():
        pass


class StudentData:

    def generateStudentData():
        pass
